File: 338.counting-bits.cs
// @leet start
public class Solution {
    public int[] CountBits(int n) {
      int[] arr = new int[n];
      for(int i=1;i<=n;i++){
        if (i % 2 == 0) {
          arr[i] = arr[i/2];
        }
        else {
          arr[i] = 1 + arr[i-1];
        }
      }
      return arr;
    }
}
// @leet end

File: 338.counting-bits.js
// @leet start
/**
 * @param {number} n
 * @return {number[]}
 */
var countBits = function(n) {
  let ans = new Array(n + 1).fill(0)
  let offset = 1
  for (let i = 1; i <= n; i++) {
    if (i === offset * 2) offset *= 2
    ans[i] = 1 + ans[i - offset]
  }
  return ans
};
// @leet end
